neo/rawio/axonarawio.py

```diff
@@ -197,51 +327,165 @@ def _get_analogsignal_chunk(self, block_index, seg_index, i_start, i_stop,
 
         # Read one channel at a time
         raw_signals = np.ndarray(shape=(num_samples,
-                                 len(channel_indexes)),
-                                 dtype=self.data_dtype)
+                                        len(channel_indexes)),
+                                 dtype=bin_dict['data_type'])
 
         for i, ch_idx in enumerate(channel_indexes):
-
             chan_offset = self.channel_memory_offset[ch_idx]
             raw_signals[:, i] = self._raw_signals[sig_ids + chan_offset]
 
         return raw_signals
 
+    def _spike_count(self, block_index, seg_index, unit_index):
+        tetrode_id = unit_index
+
+        raw_spikes = self._raw_spikes[tetrode_id]
+        nb_tetrode_spikes = raw_spikes.shape[0]
+        # also take into account last, potentially incomplete set of spikes
+        nb_unit_spikes = int(np.ceil(nb_tetrode_spikes / 4))
+        return nb_unit_spikes
+
+    def _get_spike_timestamps(self, block_index, seg_index, unit_index,
+                              t_start, t_stop):
+        assert block_index == 0
+        assert seg_index == 0
+
+        tetrode_id = unit_index
+        raw_spikes = self._raw_spikes[tetrode_id]
+
+        # spike times are repeated for each contact -> use only first contact
+        unit_spikes = raw_spikes['spiketimes'][::4]
+
+        # slice spike times only if needed
+        if t_start is None and t_stop is None:
+            return unit_spikes
+
+        if t_start is None:
+            t_start = self._segment_t_start(block_index, seg_index)
+        if t_stop is None:
+            t_stop = self._segment_t_stop(block_index, seg_index)
+
+        mask = self._get_temporal_mask(t_start, t_stop, tetrode_id)
+
+        return unit_spikes[mask]
+
+    def _rescale_spike_timestamp(self, spike_timestamps, dtype):
+        spike_times = spike_timestamps.astype(dtype)
+        spike_times /= self._to_hz(self.file_parameters['unit']['timebase'])
+        return spike_times
+
+    def _get_spike_raw_waveforms(self, block_index, seg_index, unit_index,
+                                 t_start, t_stop):
+        assert block_index == 0
+        assert seg_index == 0
+
```

Comment:
Same idea here self._raw_spike should be a 2D array.

Reply:
changed to 2D implementation